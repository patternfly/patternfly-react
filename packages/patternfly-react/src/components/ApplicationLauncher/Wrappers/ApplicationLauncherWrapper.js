import React from 'react';
import PropTypes from 'prop-types';
import ApplicationLauncher from '../ApplicationLauncher';
import ApplicationLauncherItem from '../ApplicationLauncherItem';
import { noop } from '../../../common/helpers';

const ApplicationLauncherWrapper = ({ apps, noIcons, grid, tooltip, tooltipPlacement, toggleLauncher, open }) => {
  const renderApps = apps.map((app, i) => (
    <ApplicationLauncherItem
      key={i}
      icon={app.icon}
      title={app.title}
      tooltip={app.tooltip}
      tooltipPlacement={tooltipPlacement}
      onClick={app.onClick}
      noIcons={noIcons}
    />
  ));

  return (
    <ApplicationLauncher
      grid={grid}
      tooltip={tooltip}
      tooltipPlacement={tooltipPlacement}
      open={open}
      toggleLauncher={toggleLauncher}
    >
      {renderApps}
    </ApplicationLauncher>
  );
};
ApplicationLauncherWrapper.propTypes = {
  /** Array of App Objects */
  apps: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      icon: PropTypes.string,
      tooltip: PropTypes.string,
      onClick: PropTypes.func
    })
  ),
  /** No Icons Bool */
  noIcons: PropTypes.bool,
  /** Grid instead of List (List is Default) */
  grid: PropTypes.bool,
  /** Toggle Tooltip */
  tooltip: PropTypes.string,
  /** Tooltip Placement */
  tooltipPlacement: PropTypes.string,
  /** Launcher open bool */
  open: PropTypes.bool.isRequired,
  /** Toggle launcher func */
  toggleLauncher: PropTypes.func
};
ApplicationLauncherWrapper.defaultProps = {
  noIcons: false,
  tooltip: '',
  tooltipPlacement: 'left',
  grid: false,
  toggleLauncher: noop,
  apps: [
    {
      title: 'Royal',
      icon: 'virtual-machine',
      tooltip: 'Tooltip!',
      onClick: e => {
        e.preventDefault();
      }
    }
  ]
};

export default ApplicationLauncherWrapper;
