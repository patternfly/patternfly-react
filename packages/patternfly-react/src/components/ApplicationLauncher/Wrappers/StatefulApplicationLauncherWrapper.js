import React from 'react';
import PropTypes from 'prop-types';
import ApplicationLauncherWrapper from './ApplicationLauncherWrapper';
import enhanceWithClickOutside from 'react-click-outside';

class StatefulApplicationLauncherWrapper extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      open: false
    };
  }

  toggleLauncher = () => {
    this.setState({ open: !this.state.open });
  };

  handleClickOutside = () => {
    if (this.state.open === true) {
      this.toggleLauncher();
    }
  };

  render() {
    return (
      <ApplicationLauncherWrapper
        apps={this.props.apps}
        noIcons={this.props.noIcons}
        grid={this.props.grid}
        tooltip={this.props.tooltip}
        tooltipPlacement={this.props.tooltipPlacement}
        open={this.state.open}
        toggleLauncher={this.toggleLauncher}
      />
    );
  }
}
StatefulApplicationLauncherWrapper.propTypes = {
  /** Array of App Objects */
  apps: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      icon: PropTypes.string,
      tooltip: PropTypes.string,
      onClick: PropTypes.func
    })
  ),
  /** No Icons Bool */
  noIcons: PropTypes.bool,
  /** Grid instead of List (List is Default) */
  grid: PropTypes.bool,
  /** Toggle Tooltip */
  tooltip: PropTypes.string,
  /** Tooltip Placement */
  tooltipPlacement: PropTypes.string
};
StatefulApplicationLauncherWrapper.defaultProps = {
  noIcons: false,
  tooltip: '',
  tooltipPlacement: 'left',
  grid: false,
  apps: [
    {
      title: 'Royal',
      icon: 'virtual-machine',
      tooltip: 'Tooltip!',
      onClick: e => {
        e.preventDefault();
      }
    }
  ]
};
export default enhanceWithClickOutside(StatefulApplicationLauncherWrapper);
