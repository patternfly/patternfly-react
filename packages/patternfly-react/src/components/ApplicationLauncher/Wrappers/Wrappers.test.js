import React from 'react';
import { mount } from 'enzyme';
import ApplicationLauncherWrapper from './ApplicationLauncherWrapper';
import StatefulApplicationLauncherWrapper from './StatefulApplicationLauncherWrapper';

const handleClick = e => {
  e.preventDefault();
};

const Apps = [
  {
    title: 'Ipsum',
    icon: 'pficon pficon-home',
    tooltip: 'Tooltip!',
    onClick: e => {
      handleClick(e);
    }
  },
  {
    title: 'Royal',
    icon: 'pficon pficon-virtual-machine',
    tooltip: 'Tooltip!',
    onClick: e => {
      handleClick(e);
    }
  },
  {
    title: 'Lemon',
    icon: 'pficon pficon-storage-domain',
    tooltip: 'Tooltip!',
    onClick: e => {
      handleClick(e);
    }
  },
  {
    title: 'Domain',
    icon: 'pficon pficon-domain',
    tooltip: 'Tooltip!',
    onClick: e => {
      handleClick(e);
    }
  }
];

test('ApplicationLauncherWrapper is working properly as Grid', () => {
  const component = mount(<ApplicationLauncherWrapper apps={Apps} grid open={false} />);

  expect(component.render()).toMatchSnapshot();
});

test('ApplicationLauncherWrapper is working properly as List with no Icons', () => {
  const component = mount(<ApplicationLauncherWrapper apps={Apps} noIcons open={false} />);

  expect(component.render()).toMatchSnapshot();
});

test('ApplicationLauncherWrapper is working properly as Grid when open', () => {
  const component = mount(<ApplicationLauncherWrapper apps={Apps} grid open />);

  expect(component.render()).toMatchSnapshot();
});

test('ApplicationLauncherWrapper is working properly as List with no Icons when open', () => {
  const component = mount(<ApplicationLauncherWrapper apps={Apps} noIcons open />);

  expect(component.render()).toMatchSnapshot();
});

test('StatefulApplicationLauncherWrapper is working properly as Grid', () => {
  const component = mount(<StatefulApplicationLauncherWrapper grid apps={Apps} />);

  expect(component.render()).toMatchSnapshot();
});
