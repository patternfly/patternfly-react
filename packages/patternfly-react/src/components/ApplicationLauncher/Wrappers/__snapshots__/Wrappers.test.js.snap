// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationLauncherWrapper is working properly as Grid 1`] = `
<li
  class="applauncher-pf dropdown applauncher-pf-block-list"
>
  <button
    aria-expanded="false"
    class="nav-item-iconic dropdown-toggle btn btn-link"
    type="button"
  >
    <span
      aria-hidden="true"
      class="fa fa-th applauncher-pf-icon"
    />
    <span
      class="dropdown-title"
    >
      <span
        class="applauncher-pf-title"
      >
        Application Launcher
        <span
          aria-hidden="true"
          class="caret"
        />
      </span>
    </span>
  </button>
  <ul
    class="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-home applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Ipsum
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-virtual-machine applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Royal
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-storage-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Lemon
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Domain
        </span>
      </a>
    </li>
  </ul>
</li>
`;

exports[`ApplicationLauncherWrapper is working properly as Grid when open 1`] = `
<li
  class="applauncher-pf dropdown applauncher-pf-block-list open"
>
  <button
    aria-expanded="true"
    class="nav-item-iconic dropdown-toggle btn btn-link"
    type="button"
  >
    <span
      aria-hidden="true"
      class="fa fa-th applauncher-pf-icon"
    />
    <span
      class="dropdown-title"
    >
      <span
        class="applauncher-pf-title"
      >
        Application Launcher
        <span
          aria-hidden="true"
          class="caret"
        />
      </span>
    </span>
  </button>
  <ul
    class="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-home applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Ipsum
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-virtual-machine applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Royal
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-storage-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Lemon
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Domain
        </span>
      </a>
    </li>
  </ul>
</li>
`;

exports[`ApplicationLauncherWrapper is working properly as List with no Icons 1`] = `
<li
  class="applauncher-pf dropdown"
>
  <button
    aria-expanded="false"
    class="nav-item-iconic dropdown-toggle btn btn-link"
    type="button"
  >
    <span
      aria-hidden="true"
      class="fa fa-th applauncher-pf-icon"
    />
    <span
      class="dropdown-title"
    >
      <span
        class="applauncher-pf-title"
      >
        Application Launcher
        <span
          aria-hidden="true"
          class="caret"
        />
      </span>
    </span>
  </button>
  <ul
    class="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Ipsum
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Royal
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Lemon
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Domain
        </span>
      </a>
    </li>
  </ul>
</li>
`;

exports[`ApplicationLauncherWrapper is working properly as List with no Icons when open 1`] = `
<li
  class="applauncher-pf dropdown open"
>
  <button
    aria-expanded="true"
    class="nav-item-iconic dropdown-toggle btn btn-link"
    type="button"
  >
    <span
      aria-hidden="true"
      class="fa fa-th applauncher-pf-icon"
    />
    <span
      class="dropdown-title"
    >
      <span
        class="applauncher-pf-title"
      >
        Application Launcher
        <span
          aria-hidden="true"
          class="caret"
        />
      </span>
    </span>
  </button>
  <ul
    class="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Ipsum
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Royal
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Lemon
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          class="applauncher-pf-link-title"
        >
          Domain
        </span>
      </a>
    </li>
  </ul>
</li>
`;

exports[`StatefulApplicationLauncherWrapper is working properly as Grid 1`] = `
<li
  class="applauncher-pf dropdown applauncher-pf-block-list"
>
  <button
    aria-expanded="false"
    class="nav-item-iconic dropdown-toggle btn btn-link"
    type="button"
  >
    <span
      aria-hidden="true"
      class="fa fa-th applauncher-pf-icon"
    />
    <span
      class="dropdown-title"
    >
      <span
        class="applauncher-pf-title"
      >
        Application Launcher
        <span
          aria-hidden="true"
          class="caret"
        />
      </span>
    </span>
  </button>
  <ul
    class="dropdown-menu-right dropdown-menu"
    role="menu"
  >
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-home applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Ipsum
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-virtual-machine applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Royal
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-storage-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Lemon
        </span>
      </a>
    </li>
    <li
      class="applauncher-pf-item"
      role="presentation"
    >
      <a
        class="applauncher-pf-link"
        href="#"
        role="menuitem"
      >
        <span
          aria-hidden="true"
          class="pficon pficon-pficon pficon-domain applauncher-pf-link-icon"
        />
        <span
          class="applauncher-pf-link-title"
        >
          Domain
        </span>
      </a>
    </li>
  </ul>
</li>
`;
